src/Features/CSharp/Completion/CompletionProviders/EnumAndCompletionListTagCompletionProvider.cs

```diff
@@ -121,6 +121,13 @@ public override bool IsTriggerCharacter(SourceText text, int characterPosition,
 
         private INamedTypeSymbol GetCompletionListType(ITypeSymbol type, INamedTypeSymbol within, Compilation compilation)
         {
+            // PERF: None of the SpecialTypes include <completionlist> tags,
+            // so we don't even need to load the documentation.
```

Comment:
What's the expiration policy for this cache? We have a map of SymbolKey->SymbolKey, which we populate the first time we preselect a type. What if the user then changes the documentation for that type? We would need to reread the documentation but have no way of knowing when we need to.